modules/apps/portal-cache/portal-cache-ehcache-impl/src/main/java/com/liferay/portal/cache/ehcache/internal/configurator/EhcachePortalCacheManagerConfigurator.java

```diff
@@ -92,69 +90,15 @@ protected Properties parseProperties(
 		return properties;
 	}
 
-	private void _clearListenerConfigrations(
-		CacheConfiguration cacheConfiguration) {
-
-		if (cacheConfiguration == null) {
-			return;
-		}
-
-		List<?> factoryConfigurations =
-			cacheConfiguration.getCacheEventListenerConfigurations();
-
-		factoryConfigurations.clear();
-	}
-
-	private void _clearListenerConfigrations(Configuration configuration) {
-		List<?> listenerFactoryConfigurations =
-			configuration.getCacheManagerPeerListenerFactoryConfigurations();
-
-		listenerFactoryConfigurations.clear();
-
-		List<?> providerFactoryConfigurations =
-			configuration.getCacheManagerPeerProviderFactoryConfiguration();
-
-		providerFactoryConfigurations.clear();
-
-		FactoryConfiguration<?> factoryConfiguration =
-			configuration.getCacheManagerEventListenerFactoryConfiguration();
-
-		if (factoryConfiguration != null) {
-			factoryConfiguration.setClass(null);
-		}
-
-		_clearListenerConfigrations(
-			configuration.getDefaultCacheConfiguration());
-
-		Map<String, CacheConfiguration> cacheConfigurations =
-			configuration.getCacheConfigurations();
-
-		for (CacheConfiguration cacheConfiguration :
-				cacheConfigurations.values()) {
-
-			_clearListenerConfigrations(cacheConfiguration);
-		}
-	}
-
-	@SuppressWarnings("deprecation")
 	private boolean _isRequireSerialization(
-		CacheConfiguration cacheConfiguration) {
+		CacheConfiguration<?, ?> cacheConfiguration) {
```

Comment:
@dantewang the _isRequireSerialization does not look right. This method is used to tell if the instanced stored in ehcache requires serialize, as if yes, we have to include more info into the cached value to make sure deserialize work.

I do not think resourceType.requiresSerialization() means same thing.

Reply:
@tinatian The original (Ehcache 2.x) logic was solely relying on Ehcache's cache configuration. In general, it only returns true when the cache is configured to persist, or to overflow to disk/off-heap.

In Ehcache 3.x, I believe the equivalent would be resource pools' resource type. A cache can have multiple resource pools at the same time, as multiple tiers of storage, and whether serialization is needed is based on which pools are used.

See https://www.ehcache.org/documentation/3.10/tiering.html
and https://www.ehcache.org/apidocs/3.10.0/org/ehcache/config/ResourceType.html#requiresSerialization--